components/Contact.vue

```diff
@@ -57,8 +67,15 @@ export default {
     let fullname = ""
     let reason = ""
     let message = ""
```

Comment:
This can be handled easily with one state object.

```
const form = ref({})

<input v-model="form[fullname]" />
<input v-model="form[reason]" />
<input v-model="form[message]" />
```

Your state within composition API should either be a `ref` or `reactive` as well. `Ref` makes more sense here do look it up. If you don't use ref or reactive then your state won't actually change, try changing the input of these values and console logging them somewhere.

Reply:
Thought I read somewhere that setup() makes properties exposed are automatically reactive, but maybe that was a different method like data. Fixed this one, thanks!

Reply:
Actually, tried fixing it but when attempting to import `ref` or `reactive`, it says `export 'ref' was not found in 'vue'` (possibly because the vue version being used doesn't support compositional API?)

Reply:
`Thought I read somewhere that setup() makes properties exposed are automatically reactive, but maybe that was a different method like data. Fixed this one, thanks!`
- Try logging the values after changing them without ref or reactive and check yourself
`Actually, tried fixing it but when attempting to import ref or reactive, it says export 'ref' was not found in 'vue' (possibly because the vue version being used doesn't support compositional API?)`
- Yeah I apologize, the imports should come from `@nuxtjs/composition-api`

Reply:
`Your state within composition API should either be a ref or reactive as well. Ref makes more sense here do look it up. If you don't use ref or reactive then your state won't actually change, try changing the input of these values and console logging them somewhere.`


Instead of using `ref` I used `reactive`. I think `ref` only makes primitives reactive, while `reactive` makes objects and their properties reactive. When changing an object with that has been set as reactive using `ref`, the properties inside it are not reactive and don't change.